src/services/service.ts

```diff
@@ -287,6 +288,74 @@ export default class Service {
         return parsed_results
     }
 
+    protected serviceStream<T>(options: ReportOptions, pre_report_hook: PreReportHook) {
+        const query = this.buildCustomerReportQuery(options)
+
+        const call = this.streamSearchData(query)
+
+        return this.streamGenerator<T>(call)
+    }
+
+    private async *streamGenerator<T>(call: ClientReadableStream<SearchGoogleAdsStreamResponse>) {
+        let done = false
+        const accumulator: T[] = []
+
+        function createNewPromise() {
+            let resolveMe = () => {
+                console.log('this should never happen')
```

Comment:
We should throw in here, otherwise this mysterious message could show up in people's consoles and they'll have no idea what's going on.

Reply:
We can't throw as when assigning resolveMe as then the return types would be different (TypeScript, error) have removed the logs and we could keep it empty, run the risk of something going wrong and the user not being told however. Open to suggestions